src/strtree.c

```diff
@@ -172,7 +179,7 @@ static PyObject* STRtree_new(PyTypeObject* type, PyObject* args, PyObject* kwds)
       Py_INCREF(obj);
       kv_push(GeometryObject*, _geoms, obj);
       count++;
-      GEOSSTRtree_insert_r(ctx, tree, geom, (void*)i);
+      GEOSSTRtree_insert_r(ctx, tree, geom, &kv_A(_tree_indexes, i));
```

Comment:
with array managed in tree

```suggestion
GEOSSTRtree_insert_r(ctx, tree, geom, &(_tree_indexes[i]);
```